to_json.wdl

```diff
@@ -0,0 +1,237 @@
+				except KeyError:
+					pass
+				try:
+					feat_dic['cov'] = feat_obj.attributes['cov'][0]
+				except KeyError:
+					pass
+				try:
+					feat_dic['FPKM'] = feat_obj.attributes['FPKM'][0]
+				except KeyError:
+					pass
+				try:
+					feat_dic['TPM'] = feat_obj.attributes['TPM'][0]
+				except KeyError:
+					pass
+				# just to make sure that keys are strings, else json dump fails
+				feat_dic_str = {}
+				for key, value in feat_dic.items():
+					feat_dic_str[str(key)] = value
+				return feat_dic_str
+
+		########################################################################################################################################
+
+		class TSVtoJSON():
+			""" 
+			Convert TSV output from JGI ReadCounts script to JSON format 
+			to combine with functional annotation gff file
+			feat_dic['sense_read_count'] = DY's reads_cnt
+			feat_dic['antisense_read_count'] = DY's reads_cntA instead of feat_dic['FPKM'] = feat_obj.attributes['FPKM']
+			you will have to make it read the DY's output, ingest to a dic, read the functional annotation gff file using gffutils packages, and then add selected variables to the same dic and convert it to json and csv
+				
+			"""
+			def __init__(self, tsv_file_name: str, out_json_file: str):
+				self.tsv_file_name = tsv_file_name
+				self.out_json_file = out_json_file
+			
+			def tsv_json(self):
+				"""
+				Convert TSV to dictionaries for writing out to json
+				Uses pandas to read in CSV, rename columns, drop empty column, and create dictionary for json dumping. 
+				"""
+				tsv_obj = pd.read_csv(
+					self.tsv_file_name, sep="\t"
+					).drop(columns = ["locus_tag", "scaffold_accession"]
+					).rename(columns = {"img_gene_oid": "id", 
+										"img_scaffold_oid": "seqid",
+										"reads_cnt": "sense_read_count",
+										"reads_cntA": "antisense_read_count",
+										"locus_type": "featuretype"})
+				# tsv_dic = tsv_obj.to_dict(orient="records")
+				print("TSV to JSON completed")
+
+				write_json(tsv_obj.to_dict(orient="records"), self.out_json_file)
+	
+		# Function call #######################################################################################################################################
+		final_jsons(gff_in = "~{gff}", rc_in = "~{readcount}", prefix = "~{prefix}")
+		########################################################################################################################################
+		CODE
+	>>>
+
+	output{
+		File gff_json = "~{prefix}_paired.gff.json"
+        File rc_json = "~{prefix}_paired.rc.json"
+        File gff_rc_json = "~{prefix}_gff_rc.json"
+		File cds_json = "~{prefix}_cds_counts.json"
+		File sense_json = "~{prefix}_sense_counts.json"
+		File anti_json = "~{prefix}_antisense_counts.json"
+        File top100_json = "~{prefix}_top100_features.json"
+		File sorted_json = "~{prefix}_sorted_features.json"
+        File sorted_tsv = "~{prefix}_sorted_features.tsv"
+	}
+
+	runtime {
+		docker: container
```

Comment:
add memory and cpu to the runtime block

Reply:
@chienchi  can you take care of this? This was your image from 2 years ago.

Reply:
added https://github.com/microbiomedata/metaT/pull/36/commits/08cb4e1fc49c20fd041a1f6b5d61919cfb521b85